packages/snaps-simulation/src/structs.ts

```diff
@@ -143,6 +144,22 @@ export const TransactionOptionsStruct = object({
   ),
 });
 
+export const KeyringOptionsStruct = object({
+  /**
+   * The origin making the Keyring request.
+   */
+  origin: defaulted(string(), 'metamask.io'),
+  /**
+   * Keyring request object with params.
+   */
+  request: object({
+    params: JsonRpcParamsStruct,
+    id: union([string(), number(), literal(null)]),
```

Comment:
Wondering if we should make this a bit less strict and not require `jsonrpc` and `id`?

Reply:
Refactored to use `RequestOptions`